service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -94,13 +101,42 @@ public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = getCaseData(callbackRequest.getCaseDetails());
         CaseData caseDataBefore = getCaseDataBefore(callbackRequest);
 
-        if (isRemovingNewSDO(caseData, caseDataBefore)) {
+        Optional<StandardDirectionOrder> removedSDO = getRemovedSDO(caseData, caseDataBefore);
+        Optional<CaseManagementOrder> removedCMO = getRemovedCMO(caseData, caseDataBefore);
+
+        if (removedSDO.isPresent()) {
             publishEvent(new PopulateStandardDirectionsEvent(callbackRequest));
+            publishEvent(new StandardDirectionsOrderRemovedEvent(
+                caseData, removedSDO.map(StandardDirectionOrder::getRemovalReason).orElse(null)));
+        } else if (removedCMO.isPresent()) {
+            publishEvent(
+                new CMORemovedEvent(caseData, removedCMO.map(CaseManagementOrder::getRemovalReason).orElse(null)));
         }
     }
 
-    private boolean isRemovingNewSDO(CaseData caseData, CaseData caseDataBefore) {
-        return !Objects.equals(caseData.getHiddenStandardDirectionOrders(),
-            caseDataBefore.getHiddenStandardDirectionOrders());
+    private Optional<StandardDirectionOrder> getRemovedSDO(CaseData caseData, CaseData caseDataBefore) {
+        List<Element<StandardDirectionOrder>> hiddenSDOs = caseData.getHiddenStandardDirectionOrders();
+        List<Element<StandardDirectionOrder>> previousHiddenSDOs = caseDataBefore.getHiddenStandardDirectionOrders();
+
+        if (!Objects.equals(hiddenSDOs, previousHiddenSDOs)) {
+            return hiddenSDOs.stream()
+                .filter(removedSDO -> !previousHiddenSDOs.contains(removedSDO))
+                .findFirst()
+                .map(Element::getValue);
+        }
+        return Optional.empty();
+    }
```

Comment:
I think these should live in the service rather than the controller. That way we can test more as unit tests rather than integration tests.

Reply:
Moved to service and added unit tests